karyon3-core/src/main/java/com/netflix/karyon/Karyon.java

```diff
@@ -333,84 +326,51 @@ public LifecycleInjector start() {
         
         // Create the main LifecycleManager to be used by all levels
         final LifecycleManager manager = new LifecycleManager();
         
-        final KaryonFeatureSetImpl featureSet = new KaryonFeatureSetImpl(new IdentityHashMap<>(features));
-        
         // Construct the injector using our override structure
         try {
-            final Module coreModule = Modules.override(
-                ImmutableList.<Module>builder()
-                    .addAll(modules)
-                    .add(new CoreModule())
-                    .add(new CoreAdminModule())
-                    .add(new LifecycleModule())
-                    .add(new ConditionalSupportModule())
-                    .add(new AbstractModule() {
-                        @Override
-                        protected void configure() {
-                            bind(KaryonFeatureSet.class).toInstance(featureSet);
-                            bind(LifecycleManager.class).toInstance(manager);
-                            
-                            Multibinder<String> profiles = Multibinder.newSetBinder(binder(), String.class, Profiles.class);
-                            for (String profile : Karyon.this.profiles) {
-                                profiles.addBinding().toInstance(profile);
-                            }
-                            
-                            if (args != null) {
-                                bind(String[].class).annotatedWith(Arguments.class).toInstance(args);
-                            }
-                            else {
-                                bind(String[].class).annotatedWith(Arguments.class).toInstance(new String[]{});
-                            }
-                        }
-                    })
-                    .build())
-                .with(overrideModules);
+            List<Module> coreModules = new ArrayList<>();
+            coreModules.addAll(modules);
+            coreModules.add(module);
+            coreModules.add(new CoreModule());
+            coreModules.add(new AbstractModule() {
+                @Override
+                protected void configure() {
+                    bind(KaryonFeatureSet.class).toInstance(featureSet);
+                    bind(LifecycleManager.class).toInstance(manager);
+                    bind(new TypeLiteral<Set<String>>() {}).annotatedWith(Profiles.class).toInstance(profiles);
+                    bind(String[].class).annotatedWith(Arguments.class).toInstance(args);
+                }
+            });
             
-            for (Element binding : Elements.getElements(coreModule)) {
-                LOG.debug("Binding : {}", binding);
-            }
+            List<ModuleListTransformer> localTransformers = new ArrayList<ModuleListTransformer>(transformers);
+            localTransformers.add(0, new AutoBindingModuleTransformer(autoBinders));
+            localTransformers.add(new BindingLoggingModuleTransformer());
```

Comment:
I'd need to give this some thought.  The main difference between ModuleListTransformer and AutoBinder are that AutoBinder doesn't actually get invoked unless it is specifically needed for a missing binding that the AutoBinder matches.  ModuleListTransformer has no such conditions so we'd be required to always apply it.  Also, there is a more strict ordering requirement for ModuleListTransformers so we'd need to introduce something more robust than the @Priority annotation.  Perhaps build a DAG as @jhspaybar recommended.  This seems like more works than it's worth though.  Do you have a specific use case in mind for using the service loader to pull in a ModuleListTransformer?